Quote postgres privilege target names

alertedsnake opened the pull request:
### What does this PR do?

Quotes privilege update target names and role names in salt.states.postgres_privileges.
You can put certain characters in your table/database/role names which PostgreSQL
allows but then requires you to quote.  So we should always quote these.
### What issues does this PR fix or reference?

None
### Previous Behavior

Privilege target and role names were not quoted, this results in exceptions when you have a table, database, or role name which contains odd characters, such as a dash.
### New Behavior

Quotes things which need to be quoted.
### Tests written?

Unit tests have been updated.

Commit:
Quote postgres privilege target names

Postgres lets you put characters in table/database names which you then must
quote.  So we should always quote.

cachedout commented:
Hi @alertedsnake. This is breaking the unit tests for postgres. Would you mind having a look? Let me know if I can help.

alertedsnake commented:
Ooh, interesting. I tested it all ways with a live server, but didn't check
the tests. First thing in the morning, sorry about that!

cachedout commented:
@alertedsnake Many thanks!

Commit:
Updating unit tests

alertedsnake commented:
@cachedout That ought to fix the Postgres unit tests - I notice a bunch of other things fail, but nothing I touched.

Commit:
Also quote role names.

Role names can also have dashes (or others) in them, so we must also quote
them.
